### PR TITLE
IGNITE-23815 Fix assertion during RotatedIdPartRecord read from WAL

### PR DESCRIPTION
**Pull request by skorotkov**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [X] There is a single JIRA ticket related to the pull request. 
- [X] The web-link to the pull request is attached to the JIRA ticket.
- [X] The JIRA ticket has the _Patch Available_ state.
- [X] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [X] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [X] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [X] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/serializer/RecordDataV1Serializer.java [hunk 1]
```diff
@@ -1177,7 +1177,7 @@ WALRecord readPlainRecord(RecordType type, ByteBufferBackedDataInput in,
 
                 byte rotatedIdPart = in.readByte();
 
-                res = new RotatedIdPartRecord(cacheId, pageId, rotatedIdPart);
+                res = new RotatedIdPartRecord(cacheId, pageId, ((int)rotatedIdPart) & 0xFF);
```
Comment (ivandasch) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/serializer/RecordDataV1Serializer.java:1180]: Byte#toUnsignedInt

Comment (skorotkov) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/serializer/RecordDataV1Serializer.java]: Yes, fixed

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/wal/WalRotatedIdPartRecordTest.java [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        disableCheckpoints(grid(0));
+        disableCheckpoints(grid(1));
+
+        IgniteInternalFuture<?> touchFut = runAsync(() -> {
+            while (!stop.get())
+                cln.cache(DEFAULT_CACHE_NAME).query(new ScanQuery<>()).forEach((v) -> {});
+        }, "touch");
+
+        try {
+            touchFut.get(120, TimeUnit.SECONDS);
```
Comment (alex-plekhanov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/wal/WalRotatedIdPartRecordTest.java:159]: Perhaps getTestTimeout() milliseconds?

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/wal/WalRotatedIdPartRecordTest.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+                /** {@inheritDoc} */
+                @Override public <T> @Nullable T createComponent(PluginContext ctx, Class<T> cls) {
+                    if (IgniteWriteAheadLogManager.class.equals(cls))
+                        return (T)new TestFileWriteAheadLogManager(((IgniteEx)ctx.grid()).context(), stop);
+
+                    return null;
+                }
+            });
+
+            cfg.setMetricsLogFrequency(1000);
```
[the remark this answers is not in the document]
Comment (skorotkov) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/wal/WalRotatedIdPartRecordTest.java:89]: I added it to look for size of striped pool queue size during test debug.

Will remove

**Review by alex-plekhanov — APPROVED** [review]

**Pull request closed without merging**